In [23]:
import pandas as pd

unwired = pd.read_excel('Data/20103_4akal.xlsx')

# Remove search irrelevant data
unwired = unwired[[
    'Site_Mode',
    'EIRP',
    'EIRP_Unit',
    'Antenna_Ht',
    'Site_Name',
    'Location',
    'State',
    'Antenna_Type',
    'Antenna_Gain',
    'Lic_BW',
    'Licence_Sub_Type',
    'Freq_Upper',
    'Freq_Lower']]

# Filter out Rx entries
unwired = unwired[unwired['Site_Mode'] == 'T']

# Hz to MHz
unwired['Lic_BW'] = (unwired['Lic_BW'] / 1000000).round(3)
unwired['Freq_Upper'] = (unwired['Freq_Upper'] / 1000000).round(7)
unwired['Freq_Lower'] = (unwired['Freq_Lower'] / 1000000).round(7)
unwired.rename(columns={'Lic_BW':'BW (MHz)'}, inplace=True)

# 538 Tx entires
unwired.reset_index(drop=True, inplace=True)
unwired.to_excel('Data/unwired.xlsx')

# Bandwidth Stats
BWstats = unwired['BW (MHz)'].value_counts()
print(BWstats)

# Antenna Gain Stats
AntennaGain = unwired['Antenna_Gain'].describe
print(AntennaGain)

5.0     336
14.0     81
7.0      76
2.0      27
3.5      10
10.5      8
Name: BW (MHz), dtype: int64
<bound method NDFrame.describe of 0      16.0
1      16.0
2      16.0
3      16.0
4      16.0
       ... 
533    16.0
534    16.0
535    16.0
536    16.0
537    16.0
Name: Antenna_Gain, Length: 538, dtype: float64>


## Summary of Unwired Deployments

Summary of historical 

Total Tx entires = 538

Bandwidth Distributions:

| Bandwidth | No. of Devices | Percentage of Devices |
| --------- | -------------- | --------------------- |
| 2         | 27             | 5.02 %                |
| 3.5       | 10             | 5.02 %                |
| 5         | 336            | 5.02 %                |
| 7         | 76             | 5.02 %                |
| 10.5      | 8              | 5.02 %                |
| 14        | 81             | 5.02 %                |